### ISSUE TITLE
[Android] Reset rewards doesn't work just turns off rewards

### ISSUE THREAD

**Issue by srirambv** [brave/brave-browser#8626]
## Description 
Reset rewards doesn't work just turns off rewards

## Steps to reproduce 

   1. Clean install 1.5.110
   2. Enable rewards via onboarding or via panel or via chrome://rewards
   3. Open settings -> Advanced -> Brave Rewards -> Reset rewards
   4. Rewards doesn't reset instead just turns off rewards

## Actual result 
![reset-rewards](https://user-images.githubusercontent.com/17010094/76382865-0b9ca380-6380-11ea-9e52-d35e494d045f.gif)

## Expected result
Rewards should be reset instead of turning it off

## Issue reproduces how often 
Easy

## Issue happens on 
   - Current Play Store version? No
   - Beta build? Yes

## Device details
   - Install type (ARM, x86): ARM
   - Device (Phone, Tablet, Phablet): Phone 
   - Android version: 10
   
## Brave version
1.5.110 Stable x64 build

### Website problems only
- Does the issue resolve itself when disabling Brave Shields?  NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

### Additional information

cc: @SergeyZhukovsky @gdregalo @samartnik @bbondy 
Same issue on upgrade as well from tabs -> core

**Comment by emerick** [comment]
I believe I've seen this before and it was due to `ResetOnFilesTaskRunner` failing. It seemed to happen because we requested a non-recursive delete of the `rewards_service` directory, but `confirmations.json` was still in the directory and thus the function returned `false`.

Either `confirmations.json` shouldn't be appearing there or it needs to be included in the list of files to delete on reset, I'm not sure which. cc: @NejcZdovc for advice here.
